### PR TITLE
integrated evaluated flag into metrics calculations

### PR DESCRIPTION
**Pull request by dhuber23b**
Any object that has its evaluated metric set to false (i.e., outside the room) should not count against submission scores.  However, these objects still need to be included in the data association step because the submitter may detect objects that are outside the room and model them.

### CODE REVIEW

File: sumo-api/sumo/metrics/evaluator.py [hunk 1]
```diff
@@ -282,7 +282,8 @@ def shape_score(self):
         float - shape similarity score
         """
 
-        n_gt = len(self._ground_truth.elements)
+        # number of ground truth elements with evaluated == True
+        n_gt = sum(v.evaluated == True for v in self._ground_truth.elements.values())
```
Comment (dellaert) [inline sumo-api/sumo/metrics/evaluator.py]: Here and in a few other places you say `v.evaluated == True`, but you should just say `v.evaluated`

File: sumo-api/sumo/metrics/evaluator.py [hunk 2]
```diff
@@ -291,12 +292,19 @@ def shape_score(self):
             # det_matches = 1 if correct detection, 0 if false positive
             det_matches = []
```
Comment (dellaert) [inline sumo-api/sumo/metrics/evaluator.py]: Clearly should be a list of tuples generated by a list comprehension, but I won't nitpick too much...

**Comment by dhuber23b** [comment]
> I don't completely understand what goes on, a comment on the PR could be useful next time

Are you able to see the comment that is already on the PR?
"Any object that has its evaluated metric set to false (i.e., outside the room) should not count against submission scores. However, these objects still need to be included in the data association step because the submitter may detect objects that are outside the room and model them."

File: sumo-api/sumo/metrics/evaluator.py (hunk 2 above)
Comment (dhuber23b) [reply sumo-api/sumo/metrics/evaluator.py]: True, but the code is more readable with it written out.  Plus, the 80 column limit on code makes long, complicated list comprehensions even harder to understand.

File: sumo-api/sumo/metrics/evaluator.py (hunk 1 above)
Comment (dhuber23b) [reply sumo-api/sumo/metrics/evaluator.py]: done.

**Pull request merged**